client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.tsx

```diff
@@ -187,8 +195,40 @@ export function MapVehiclesSidebar({
     );
     const endStop = useMemo(() => stops[stops.length - 1], [stops]);
 
+    const allStops = useSelector((state: State) => state.publicTransport.stops);
+    const setSelectedStop = useCallback(
+        (stopId: string) => {
+            const stop = allStops.find((stopFullData) => stopFullData.attributes.stopId === stopId);
+
+            if (!stop) {
+                return;
+            }
+
+            const { attributes: stopData } = stop;
+
+            dispatch(
+                setCurrentStop({
+                    currentStop: stopId,
+                    shouldClear: false,
+                }),
+            );
+
+            sidebarService.open({
+                component: <MapStopsSidebar type={stopData.type} name={stopData.title} />,
+                onClose: () => dispatch(setCurrentStop(null)),
+            });
+        },
+        [dispatch],
+    );
+
     return (
-        <div className={cn(styles.MapVehiclesSidebar)}>
+        <div
+            className={cn(styles.MapVehiclesSidebar)}
+            style={{
+                // @ts-ignore
+                '--vehicle-color': VEHICLE_TYPE_COLORS[type],
+            }}
```

Comment:
Let's remove `// @ts-ignore`

```suggestion
            style={{ '--vehicle-color': VEHICLE_TYPE_COLORS[type] } as React.CSSProperties}
```